File: src/collections/News.ts
import type { CollectionConfig } from 'payload'

export const News: CollectionConfig = {
  slug: 'news',
  admin: {
    useAsTitle: 'title',
    defaultColumns: ['title', 'category', 'source', 'publishDate', 'isFeatured'],
  },
  access: {
    read: () => true,
  },
  fields: [
    {
      name: 'title',
      type: 'text',
      required: true,
    },
    {
      name: 'source',
      type: 'text',
      required: true,
    },
    {
      name: 'category',
      type: 'select',
      required: true,
      options: [
        { label: 'Economy', value: 'Economy' },
        { label: 'Investment', value: 'Investment' },
        { label: 'Events', value: 'Events' },
        { label: 'Business', value: 'Business' },
        { label: 'Policy', value: 'Policy' },
      ],
    },
    {
      name: 'summary',
      type: 'textarea',
      required: true,
    },
    {
      name: 'content',
      type: 'richText',
    },
    {
      name: 'image',
      type: 'upload',
      relationTo: 'media',
    },
    {
      name: 'url',
      type: 'text',
      required: true,
      admin: {
        description: 'External link to full article',
      },
    },
    {
      name: 'tags',
      type: 'array',
      fields: [
        {
          name: 'tag',
          type: 'text',
        },
      ],
    },
    {
      name: 'publishDate',
      type: 'date',
      required: true,
      defaultValue: () => new Date().toISOString(),
      admin: {
        date: {
          pickerAppearance: 'dayAndTime',
        },
      },
    },
    {
      name: 'isFeatured',
      type: 'checkbox',
      defaultValue: false,
    },
  ],
}

File: lib/src/domain/entities/news.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'news.freezed.dart';

/// Domain entity representing a news article.
@freezed
class News with _$News {
  const factory News({
    required String id,
    required String title,
    required String source,
    required String category,
    required String summary,
    required DateTime publishDate,
    required String url,
  }) = _News;
}

/// Available news categories.
enum NewsCategory {
  economy('Economy'),
  investment('Investment'),
  events('Events'),
  business('Business'),
  policy('Policy');

  const NewsCategory(this.label);
  final String label;

  static NewsCategory? fromString(String value) {
    return NewsCategory.values.firstWhere(
      (e) => e.label.toLowerCase() == value.toLowerCase(),
      orElse: () => NewsCategory.economy,
    );
  }
}
